### PR TITLE
Clean up class names

### PR DESCRIPTION
**Pull request by zigen**
# TL;DR
* introduce clang-tidy and [its config](https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html)
* use CamelCase [class names](https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html#cmdoption-arg-classcase)
* remove unused struct names

### CODE REVIEW

**Comment by zigen** [comment]
I want to push one more commit to enable clang-tidy check on travis CI. pls wait a moment

**Review by rdviii — APPROVED** [review]

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#  - key:             readability-identifier-naming.ConstantParameterPrefix
+#    value:           ''
+#  - key:             readability-identifier-naming.TypedefCase
+#    value:           CamelCase
+#  - key:             readability-identifier-naming.VariableCase
+#    value:           lower_case
+#  - key:             readability-identifier-naming.ParameterCase
+#    value:           lower_case
+#  - key:             readability-identifier-naming.FunctionCase
+#    value:           camelBack
```
Comment (zigen) [inline .clang-tidy:63]: I set many options to enforce the naming, but I won't fix this now because I want to keep each PR small.

Comment (rdviii) [reply .clang-tidy:63]: OK

**Comment by rdviii** [comment]
zigen's cleanup of class names.

**Pull request merged**